proposals/0012-introduce-reactNativeMetadata.md

```diff
@@ -0,0 +1,467 @@
+
+The setup would look as follows for both apps and libraries:
+
+```json
+{
+  "reactNativeManifest": {
+    "capabilities": {
+      "newArchitecture": {
+        "enabled": true
+      }
+    }
+  }
+}
+```
+
+This section will allow also for split configuration between platforms:
+
+```json
+{
+  "reactNativeManifest": {
+    "android": {
+      "capabilities": {
+        "newArchitecture": {
+          "enabled": true
+        }
+      }
+    },
+    "ios": {
+      "capabilities": {
+        "newArchitecture": {
+          "enabled": true
+        }
+      }
+    }
+  }
+}
+```
+
+If omitted, the `newArchitecture.enabled` would have a value of `false` (this may change in the future).
+
+Notice that **libraries** might be supporting both architecture. So, the `capabilities` field can also support an `oldArchitecture` field of the same shape of the `newArchitecture`. The following example shows how to use both properties.
+
+```json
+"reactNativeManifest": {
+  "capabilities": {
+    "newArchitecture": {
+      "enabled": true
+    },
+    "oldArchitecture": {
+      "enabled": true
+    }
+  }
+}
+```
+
+If omitted, the `oldArchitecture.enabled` would have a value of `true` (this may change in the future), to simplify the adoption of the manifest. In fact, the vast majority of React Native libraries are now supporting the Old Architecture and only a subset support both of them.
+
+The `oldArchitecture` property is **ignored** by apps: this is needed to avoid ambiguous setup where an app declares both architectures as enabled.
+In other words, the `newArchitecture` property has always precedence in an app.
+
+The precedence rules are that platform-specific settings override general settings. So, for example:
+
+```json
+{
+  "reactNativeManifest": {
+    "capabilities": {
+      "newArchitecture": {
+        "enabled": true
+      }
+    },
+    "android": {
+      "capabilities": {
+        "newArchitecture": {
+          "enabled": false
+        }
+      }
+    },
+  }
+}
+```
+
+would means that, in general, the app/library support the New Architecture, but not for Android.
+
+##### Planning forward: what happen when the New Architecture will be the default?
+
+The manifest is versioned, and we will use the versioning system to track breaking changes and different semantics.
+
+For version `1.0.0` the semantic would be:
+* `newArchitecture` default value set to `false`
+
+In a future version, let's say `K.0.0`, the semantic would be:
+* `newArchitecture` default value set to `true`
+* `newArchitecture` field deprecated
+
+In a more future version, let's say `N.0.0` (where N > K), the semantic would be:
+* `newArchitecture` field removed, together with the Old Architecture code.
+
+Notice that this system can also help us to track unmaintained/old packages: they would either not have the manifest at all or have old version of the manifest with its version's semantic.
+
+#### `codegenConfig` support
```

Comment:
Nit: Since all of the `capabilities` values are essentially config, I believe it make sense to simplify this name and make it just `codegen`?

```suggestion
#### `codegen` support
```